src/containers/BooksForm.js

```diff
@@ -1,19 +1,83 @@
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable import/no-cycle */
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createBook } from '../actions';
+import { randomId } from '../index';
 
-export default function BooksForm() {
-  const bookCategories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
-  return (
-    <div>
-      <form>
-        <input type="text" placeholder="Enter a new book" />
-        <select>
-          {
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: randomId(),
+      title: '',
+      category: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    if (event.target.tagName === 'INPUT') {
+      this.setState({
+        title: event.target.value,
+      });
+    } else {
+      this.setState({
+        category: event.target.value,
+      });
+    }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.setState({
+      id: randomId(),
+    });
+    if (this.state.title !== '') {
+      this.props.addBook(this.state);
+    }
+    this.setState({
+      title: '',
+      category: '',
+    });
+  }
```

Comment:
- You might want to fall back to default category on line `43`
- you might want to reset the forms after a submit
`event.target.reset();`